### PR TITLE
Use git to check remote in check_remote_has_version

### PR DESCRIPTION
**Pull request by david-nishi**
## Major changes
- use git and git-annex to determine if a tag was the latest export to a remote repo

### CODE REVIEW

File: services/datalad/datalad_service/tasks/publish.py [hunk 1]
```diff
@@ -168,22 +168,40 @@ def export_all_tags(store, dataset, cookies, get_realm, check_should_export, esL
...
+                tag,
+            ]
+        )
         info = json.loads(response[0])
         remotes = info.get('repositories containing these files', [])
         s3_PUBLIC_remote = [r for r in remotes if r.get('description') == f'[{remote}]']
-        return bool(s3_PUBLIC_remote)
-    except AttributeError as err:
-        logger.error(err)
+        remote_id_A = s3_PUBLIC_remote and s3_PUBLIC_remote[0].get('uuid')
```
Comment (nellh) [inline services/datalad/datalad_service/tasks/publish.py]: This isn't necessarily the public remote right?

File: services/datalad/datalad_service/tasks/publish.py [hunk 2]
```diff
@@ -168,22 +168,40 @@ def export_all_tags(store, dataset, cookies, get_realm, check_should_export, esL
...
+        log_remote_id_pattern = re.compile(f':(.+) .+$')
+        match = log_remote_id_pattern.search(response)
+        remote_id_B = match.group(1)
+        log_tree_id_pattern = re.compile(f'.* (.+)$')
+        match = log_tree_id_pattern.search(response)
+        tree_id_A = match.group(1)
+
+        # extract git tree id of <tag> with `git show -s <tag>`
+        response = dataset.repo.call_git(['show', '-s', '--pretty=raw', tag])
+        tree_id_pattern = re.compile(f'^tree (.+)$', re.MULTILINE)
```
Comment (nellh) [inline services/datalad/datalad_service/tasks/publish.py]: The regular expressions don't need to be format strings.

**Review by nellh — APPROVED** [review]
Couple minor comments but this logic looks correct and worked with an existing and failed export.

File: services/datalad/datalad_service/tasks/publish.py (hunk 1 above)
Comment (david-nishi) [reply services/datalad/datalad_service/tasks/publish.py]: changed to `remote_repo`

**Pull request merged**